### PR TITLE
Importer costbasis improvements

### PR DESCRIPTION
**Pull request by peloriante**
- Coinbase exports provide costbasis/proceeds information in USD, so we use that to pass on directly to costbasis
- CoinbasePro provides fee and item pricing information for each trade, so we use that to derive USD prices and pass that on for costbasis
- Kraken only provides fee in their exports (it doesn't give a total cost for trades) so we can only price the fee in USD, which we do and pass on to costbasis
- Add new Coinbase TransactionHistory importer and use that by default
- Add more expressiveness when debugging costbasis drawdowns and lot creation

### CODE REVIEW

File: perfi/transaction/chain_to_ledger.py [hunk 1]
```diff
@@ -391,6 +407,7 @@ def fee_from_debank(self, raw_data):
 
             if self.amount > 0:
                 # Use DeBank's USD price
+                # Costbais code downstream wants price_usd to represent the unit price, so we divide total gas fee by amount here
```
Comment (lhl) [inline perfi/transaction/chain_to_ledger.py:410]: typo: Costbasis

File: tests/integration/test_imports_from_exchanges.py [hunk 2]
```diff
@@ -547,6 +554,215 @@ def test_converted_from_to(self, test_db):
         assert actual_in == expected_in
 
 
+COINBASE_TRANSACTIONS_HISTORY_HEADER = """"You can use this transaction report to inform your likely tax obligations. For US customers, Sells, Converts, Rewards Income, Coinbase Earn transactions, and Donations are taxable events. For final tax obligations, please consult your tax advisor."
+
+
+
+Transactions
+User,gabe@avantbard.com,530c49d9e612ded0020002b8
```
Comment (lhl) [inline tests/integration/test_imports_from_exchanges.py:562]: PII

File: tests/integration/test_imports_from_exchanges.py [hunk 3]
```diff
@@ -547,6 +554,215 @@ def test_converted_from_to(self, test_db):
...
+        )
+        update_entity_transactions(entity_name)
+        tx_ledgers = get_tx_ledgers(test_db, "import.coinbase", address)
+
+        assert len(tx_ledgers) == 1  # only 1 ledger for a receive
+
+        expected_receive = TxLedger(
+            chain="import.coinbase",
+            address=address,
+            hash="import.coinbase_1488610341_Receive_btc",
```
Comment (lhl) [inline tests/integration/test_imports_from_exchanges.py:595]: fuzz all ids

**Review by lhl — CHANGES_REQUESTED** [review]

**Comment by lhl** [comment]
Once scrubbing PII from checkin history, remember to scrub this pull req

File: tests/integration/test_imports_from_exchanges.py (hunk 3 above)
Comment (peloriante) [reply tests/integration/test_imports_from_exchanges.py:593]: 1488610341  is a timestamp

File: perfi/transaction/chain_to_ledger.py (hunk 1 above)
Comment (peloriante) [reply perfi/transaction/chain_to_ledger.py]: fixed

File: perfi/costbasis.py [hunk 4]
```diff
@@ -1764,9 +1757,9 @@ def get_lots(self, tx, algorithm="hifo"):
                  {'AND chain = ?' if not asset_price_id else ''}
         """
         if algorithm == "hifo":
-            sql += "ORDER BY cast(price_usd as float) DESC, timestamp ASC"
```
[the remark this answers is not in the document]
Comment (peloriante) [reply perfi/costbasis.py]: fixed

File: perfi/costbasis.py [hunk 5]
```diff
@@ -1764,9 +1757,9 @@ def get_lots(self, tx, algorithm="hifo"):
...
+            sql += "ORDER BY price_usd DESC, timestamp ASC"
         if algorithm == "low":
-            sql += "ORDER BY cast(price_usd as float) ASC"
```
[the remark this answers is not in the document]
Comment (peloriante) [reply perfi/costbasis.py]: fixed

**Pull request merged**